stbt_rig.py

```diff
@@ -545,6 +553,65 @@ def cmd_download(args, portal):
         result.download_artifacts(args.artifacts or ["*"], args.artifacts_dest)
 
 
+def cmd_encrypt_secret(args, portal):
+    import base64
+
+    pubkey = portal.get_secrets_pubkey()
+    with tempfile.NamedTemporaryFile() as f:
+        f.write(pubkey.encode())
+        f.flush()
+        encrypted = subprocess.check_output(
+            ["openssl", "pkeyutl", "-inkey", f.name, "-pubin", "-encrypt"],
+            input=args.value.encode())
+    encoded = base64.b64encode(encrypted).decode()
+
+    with open("%s/.stbt.conf" % find_test_pack_root()) as f:
+        cfg = list(f)
+    _modify_config(cfg, args.key, encoded)
+    with open("%s/.stbt.conf" % find_test_pack_root(), "w") as f:
+        f.write("".join(cfg))
+
+    sys.stderr.write(
+        "Encrypted and added the secret to your stbt.conf.  The unencrypted "
+        "secret can be read during a test run with:\n"
+        "\n"
+        "    stbt.get_config('secrets', %r)\n" % args.key)
+
+
+def _modify_config(cfg, key, value):
+    import re
+
+    section_start = None
+
+    cfg_line = "%s = %s\n" % (key, value)
+
+    for n, line in enumerate(cfg):
+        if re.match(r"\s*\[\s*encrypted_secrets\s*\]", line):
+            section_start = n
+            if not line.endswith("\n"):
+                cfg[n] = line + "\n"
+            break
+
+    if section_start is None:
+        cfg.extend(["\n", "[encrypted_secrets]\n", cfg_line])
+        return
+
+    for n, line in enumerate(cfg[section_start + 1:], section_start + 1):
+        if re.match(re.escape(key) + r"\s*=", line):
+            cfg[n] = cfg_line
+            return
+        if line.strip().startswith("["):
+            section_end = n
+            break
+    else:
+        section_end = len(cfg)
+
+    for section_end in range(section_end, section_start, -1):
+        if cfg[section_end - 1].strip():
+            break
+    cfg.insert(section_end, cfg_line)
```

Comment:
We could sort the entries by name with something like `sort(cfg[start:end])`, or at least insert the new key in a position that would preserve sorting if they were already sorted.

Reply:
Implemented.  I don't know how people will want to organise their `stbt.conf`s, but they can always rearrange themselves.